.github/docs/InteractiveMessageActionButtonImplementation.md

```diff
@@ -0,0 +1,455 @@
+# Implementation of action buttons in interactive message list picker/panel
+
+## Table of content
+  - [Context](#context)
+  - [Configuration and new fields explanation:](#configuration-and-new-fields-explanation)
+    - [Update for StartChatContact request](#update-for-startchatcontact-request)
+    - [Required fields](#required-fields)
+  - [Lambda response template](#lambda-response-template)
+  - [Operation flow](#operation-flow)
+
+## Context
+
+Currently the max number of list picker options that return from server(via Lex) is 6(10 for panel picker), but our customers would like to provide more options to their end customers.
+In order to achieve this goal, we have implemented the idea of "Show more" and "Previous options" buttons in our interactive message UI. These changes will require customers to provide some new data when they configure the interactive message so that they are able to use this new feature.
+
+## Configuration and new fields explanation:
+
+*Note: `action button` is referred to the buttons with functionality of `Show more` or `Previous options` because any text can be used as their titles.*
+
+### Update for StartChatContact request
+
+In order to use this new feature, you need to have your StartChatContact API request body update. You need to add a new content type: `application/vnd.amazonaws.connect.message.interactive.response` to `SupportedMessagingContentTypes`. For example:
+
+```
+PUT /contact/chat HTTP/1.1
+Content-type: application/json
+{
+   "Attributes": { 
+      "string" : "string" 
+   },
+   "ChatDurationInMinutes": number,
+   "ClientToken": "string",
+   "ContactFlowId": "string",
+   "InitialMessage": { 
+      "Content": "string",
+      "ContentType": "string"
+   },
+   "InstanceId": "string",
+   "ParticipantDetails": { 
+      "DisplayName": "string"
+   },
+   "PersistentChat": { 
+      "RehydrationType": "string",
+      "SourceContactId": "string"
+   },
+   "RelatedContactId": "string",
+   "SupportedMessagingContentTypes": [ "application/vnd.amazonaws.connect.message.interactive.response" ] <--- Add contentType here
```

Comment:
nit: this might be confusing because customers might have included "text/plain", "text/markdown". Can we add ellipses to indicate that other values can be present in the array as well? like 

```
[ ... , "application/vnd.amazonaws.connect.message.interactive.response" ]
```

Reply:
oh, good point, yeah I will update this.